index.js

```diff
@@ -68,28 +112,14 @@ class ExecutorRouter extends Executor {
      * @return {Promise}
      */
     _start(config) {
-        const annotations = config.annotations || {};
-        const executorType = annotations[ANNOTATION_EXECUTOR_TYPE];
-        const executor = this[executorType] || this.defaultExecutor; // Route to executor (based on annotations) or use default executor
-
-        return executor.start(config);
-    }
+        const options = this.getFullConfig(config);
```

Comment:
Nothing is going to prevent a caller from invoking this function outside of a Promise, since private functions don't really exist in the JS realm. In addition, the function documentation states that it always returns a Promise.